src/utils/fetch-source.js

```diff
@@ -89,15 +99,68 @@ export async function fetchSource(
       }
       break;
     }
+
     case 'CORS_ENABLED_ARCHIVE': {
-      // Not handled yet.
-      errors.push({ type: 'NO_KNOWN_CORS_URL' });
+      // Download the archive, and then get the file from the archive.
+      // This is the path for source code from crates.io: We download the entire
+      // crate, which is a .tar.gz file, and then we store the uncompressed bytes
+      // in the archive cache.
+      // We only download each archive once, and reuse the stored bytes when we
+      // want to show other files from the same archive.
+
+      const { archiveUrl: url, pathInArchive } = downloadRecipe;
+      let promise: Promise<Uint8Array> | void = archiveCache.get(url);
+
+      if (promise === undefined) {
+        // Create a promise to load the archive, but don't await it yet.
+        promise = (async () => {
+          const response = await callbacks.fetchUrlResponse(url);
+          const bytes = new Uint8Array(await response.arrayBuffer());
+          return isGzip(bytes) ? decompress(bytes) : bytes;
+        })();
+        archiveCache.set(url, promise);
+      } else {
+        // TODO: callbacks.onWaitingForUrlWhichIsAlreadyBeingFetched(url)
```

Comment:
I don't feel this TODO would be necessary, as we already have a useful message in the UI. If you think it's necessary, then maybe file a follow-up? Otherwise just remove this else block :-)

Reply:
We don't have a message in the UI in the case where this else block is reached:

 1. Double-click one function from the hashbrown-0.11.0 crate.
 2. Before the network load completes, double-click a function in a different file from the same crate.

This currently doesn't display a message, because we don't call `fetchUrlResponse` on the second file (because an archiveCache entry already exists), so we also don't dispatch a "begin loading" action.

Reply:
I see, thanks